openhtf/util/logs.py

```diff
@@ -199,40 +200,48 @@ def emit(self, record):
     self._notify_update()
 
 
+class CliFormatter(logging.Formatter):
+  """Formats log messages for printing to the CLI."""
+  def format(self, record):
+    """Format the record as tersely as possible but preserve info."""
+    super(CliFormatter, self).format(record)
+    localized_time = datetime.datetime.fromtimestamp(record.created)
+    terse_time = localized_time.strftime(u'%H:%M:%S')
+    terse_level = record.levelname[0]
+    terse_name = record.name.split('.')[-1]
+    if record.name.startswith(RECORD_LOGGER_PREFIX):
+      # The [1:] split used here is to discard the extra leading dot.
+      loggers = record.name.split(RECORD_LOGGER_PREFIX)[1][1:].split('.')
+      # If nested under a test record, show where the message came from.
+      if len(loggers) > 1:
+        terse_name = '<{type}: {identifier}>'.format(
+            type=loggers[1], # First thing after UUID should be the logger type.
+            identifier=loggers[-1]) # Last thing should uniquely id the logger.
+      else:
+        # Fall back to using the last five characters of the test UUID.
+        terse_name = '<test %s>' % loggers[0][-5:]
+    return '{lvl} {time} {logger} - {msg}'.format(lvl=terse_level,
+                                                     time=terse_time,
```

Comment:
style nit: additional lines should line up with the open parenthesis.

Reply:
Oops! Had adjusted the template string and forgot to fix the lines below.